ntbs-service/wwwroot/source/Components/ValidateTravelOrVisit.ts

```diff
@@ -0,0 +1,135 @@
+import Vue from 'vue';
+import { getHeaders } from '../helpers';
+import axios from 'axios';
+
+type TravelOrVisitVariables = {
+    hasData: string,
+    totalNumberOfCountries: string,
+    country1Id: string,
+    country2Id: string,
+    country3Id: string,
+    stayLengthInMonths1: string,
+    stayLengthInMonths2: string,
+    stayLengthInMonths3: string
+};
+
+const ValidateTravelOrVisit = Vue.extend({
+    props: {
+        modelType: {
+            type: String
+        }
+    },
+    methods: {
+        prependRefWithModelType: function (ref: String) {
+            return `${this.$props.modelType}-${ref}`;
+        },
+
+        validate: function (event: FocusEvent) {
+            if (this.$el.contains(event.relatedTarget)) {
+                return;
+            }
+
+            const travelOrVisitVariables = this.getTravelOrVisitVariables();
+            const requestConfig = {
+                url: `Travel/Validate${this.$props.modelType}`,
+                headers: getHeaders(),
+                params: travelOrVisitVariables
+            }
+
+            axios.request(requestConfig)
+                .catch((error: any) => {
+                    console.log(error.response);
+                })
+                .then((response: any) => {
+                    this.resetErrors();
+                    const data = response.data;
+
+                    for (let key in data) {
+                        if (data.hasOwnProperty(key)) {
+                            // Model errors can be double counted as 'Model.Property' and 'Property'.
+                            // Here it's convenient to act on 'Property' only
```

Comment:
Is this actually needed here, or just result of copy-pasta?

Reply:
Prevents enumeration of properties from the prototype chain. It's a suggested code quality improvement by resharper.